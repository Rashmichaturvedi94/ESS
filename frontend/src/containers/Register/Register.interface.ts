export interface RegisterProps {
  myProp?: boolean;
  /* Add props here. */
}
