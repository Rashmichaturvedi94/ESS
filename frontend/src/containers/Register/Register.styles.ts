import styled from 'styled-components';

export const Register = styled.div`
  flex: 1;
  display: flex;
  flex-direction: row;
`;
