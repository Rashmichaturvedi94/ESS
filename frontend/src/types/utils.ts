export type Nullable<T> = {
  [P in keyof T]: T[P] | null;
};

export type Uri = string;

export type DateString = string;

export type LocalOrRemoteFile = File | Uri;

export type Modify<T, R> = Omit<T, keyof R> & R;
