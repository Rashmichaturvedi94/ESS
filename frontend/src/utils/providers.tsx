import React, { ComponentType, FC } from 'react';
import * as Sentry from '@sentry/react';
import { QueryClientProvider } from 'react-query';
import DateFnsUtils from '@date-io/date-fns';
import { ThemeProvider } from 'styled-components';
import { StylesProvider, MuiThemeProvider } from '@material-ui/core';
import { MuiPickersUtilsProvider } from '@material-ui/pickers';
import { queryClient } from '../api';
import { theme } from '../styles/theme';


const createProviderHOC = (Provider: ComponentType) => <P extends object>(
  Component: ComponentType<P>,
) => (props: P) => (
  <Provider>
    <Component {...props} />
  </Provider>
);
const FallbackComponent = () => <div>An error has occurred</div>;

export const Providers: FC = ({ children }) => (
  <Sentry.ErrorBoundary fallback={FallbackComponent} showDialog>
    <StylesProvider injectFirst>
      <MuiThemeProvider theme={theme}>
        <ThemeProvider theme={theme}>
          <MuiPickersUtilsProvider utils={DateFnsUtils}>
            <QueryClientProvider client={queryClient}>
              {children}
            </QueryClientProvider>
          </MuiPickersUtilsProvider>
        </ThemeProvider>
      </MuiThemeProvider>
    </StylesProvider>
  </Sentry.ErrorBoundary>
);

export const withProviders = createProviderHOC(Providers);
