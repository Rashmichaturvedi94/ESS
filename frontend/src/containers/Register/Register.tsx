import React, { FC } from 'react';
import { Box, Button, Typography } from '@material-ui/core';
import { useHistory } from 'react-router-dom';
import ImportContactsIcon from '@material-ui/icons/ImportContacts';
import { Register as RegisterComponent } from './Register.styles';
import { RegisterProps } from './Register.interface';
import { RegisterForm } from '../../components/RegisterForm';
import { paths } from '../../const/paths';
import { useLogin } from '../../api';

export const Register: FC<RegisterProps> = () => {
  const { error, mutate } = useLogin();
  const history = useHistory();

  return (
  <RegisterComponent>
    <Box flex={1} display="flex" alignItems="center" justifyContent="center">
      <Box style={{backgroundColor:'#C4C4C4'}} width="45%" height="100%">
        <ImportContactsIcon style={{ color: 'black', fontSize: 150, marginTop: 50, marginLeft: 150}} />
        <Typography variant="h2" component="h2" align='center'>
          Genious
        </Typography>
        <Typography variant="h3" align='center'>E-Learning</Typography>
      </Box>
      <Box display="flex" flexDirection="column" width="55%" gridRowGap={20}>
        <RegisterForm
          errors={error?.response?.data}
          onSubmit={(data) => mutate({ data })}
        />
        <Button
          fullWidth
          variant="text"
          onClick={() => history.push(paths.login)}
        >
          Already have an account? Login
        </Button>
      </Box>
    </Box>
  </RegisterComponent>
  );
};
