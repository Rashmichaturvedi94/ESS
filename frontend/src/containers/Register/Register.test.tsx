import React from 'react';
import { render } from '../../utils/testUtils';
import { Register } from './Register';

describe('components/Register', () => {
  it('should render', () => {
    const { container } = render(<Register />);
    expect(container).toMatchSnapshot();
  });
});
