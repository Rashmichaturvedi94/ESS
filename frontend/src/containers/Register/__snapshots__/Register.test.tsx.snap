// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Register should render 1`] = `
<div>
  <div
    class="sc-bdnxRM bOAqhc"
  >
    <div
      class="MuiBox-root MuiBox-root-1"
    >
      <div
        class="MuiBox-root MuiBox-root-2"
        style="background-color: rgb(196, 196, 196);"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          style="color: black; font-size: 150px; margin-top: 50px; margin-left: 150px;"
          viewBox="0 0 24 24"
        >
          <path
            d="M17.5 4.5c-1.95 0-4.05.4-5.5 1.5-1.45-1.1-3.55-1.5-5.5-1.5S2.45 4.9 1 6v14.65c0 .65.73.45.75.45C3.1 20.45 5.05 20 6.5 20c1.95 0 4.05.4 5.5 1.5 1.35-.85 3.8-1.5 5.5-1.5 1.65 0 3.35.3 4.75 1.05.41.21.75-.19.75-.45V6c-1.49-1.12-3.63-1.5-5.5-1.5zm3.5 14c-1.1-.35-2.3-.5-3.5-.5-1.7 0-4.15.65-5.5 1.5V8c1.35-.85 3.8-1.5 5.5-1.5 1.2 0 2.4.15 3.5.5v11.5z"
          />
        </svg>
        <h2
          class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter"
        >
          Genious
        </h2>
        <h3
          class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter"
        >
          E-Learning
        </h3>
      </div>
      <div
        class="MuiBox-root MuiBox-root-3"
      >
        <form>
          <div
            class="MuiBox-root MuiBox-root-4"
            style="margin: 0px 220px;"
          >
            <div
              class="MuiFormControl-root MuiTextField-root sc-dlnjwi eVtzLs MuiFormControl-fullWidth"
            >
              <div
                class="MuiInputBase-root MuiInput-root MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input"
                  name="name"
                  placeholder="Name"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root sc-hKFxyN fisPQT MuiFormControl-fullWidth"
            >
              <div
                class="MuiInputBase-root MuiInput-root MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input"
                  name="username"
                  placeholder="Username"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root sc-eCApnc Na-DNs MuiFormControl-fullWidth"
            >
              <div
                class="MuiInputBase-root MuiInput-root MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input"
                  name="email"
                  placeholder="Email"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root sc-jSFjdj kvxhpy MuiFormControl-fullWidth"
            >
              <div
                class="MuiInputBase-root MuiInput-root MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input"
                  name="password"
                  placeholder="Password"
                  type="password"
                  value=""
                />
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root sc-jSFjdj kvxhpy MuiFormControl-fullWidth"
            >
              <div
                class="MuiInputBase-root MuiInput-root MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input"
                  name="password2"
                  placeholder="Confirm Password"
                  type="password2"
                  value=""
                />
              </div>
            </div>
            <div
              aria-label="Type"
              class="MuiFormGroup-root"
              role="radiogroup"
            >
              <label
                class="MuiFormControlLabel-root sc-gKAaRy cQITCC"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiRadio-root"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <input
                      class="PrivateSwitchBase-input-8"
                      name="Type1"
                      type="radio"
                      value="Author"
                    />
                    <div
                      class="PrivateRadioButtonIcon-root-9"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-10"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                        />
                      </svg>
                    </div>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </span>
                <span
                  class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                >
                  Author
                </span>
              </label>
              <label
                class="MuiFormControlLabel-root sc-gKAaRy cQITCC"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiRadio-root"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <input
                      class="PrivateSwitchBase-input-8"
                      name="Type1"
                      type="radio"
                      value="Student"
                    />
                    <div
                      class="PrivateRadioButtonIcon-root-9"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-10"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                        />
                      </svg>
                    </div>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </span>
                <span
                  class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                >
                  Student
                </span>
              </label>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                Register
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </form>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-fullWidth"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Already have an account? Login
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
